xds/internal/testutils/fakeclient/client.go

```diff
@@ -143,14 +146,28 @@ func (xdsC *Client) WaitForWatchCluster(ctx context.Context) (string, error) {
 // Not thread safe with WatchCluster. Only call this after
 // WaitForWatchCluster.
 func (xdsC *Client) InvokeWatchClusterCallback(update xdsclient.ClusterUpdate, err error) {
-	xdsC.cdsCb(update, err)
+	// Keeps functionality with previous usage of this, if single callback call that callback.
+	if len(xdsC.cdsCbs) == 1 {
```

Comment:
Did any other tests fail without this?

I would expect `update.ClusterName` to match the name used for watch, so `xdsC.cdsCbs[update.ClusterName](update, err)` should still work. But we may not be that careful writing tests, so it's possible that you need this special handling to keep the existing tests working.

Reply:
It's been so long since I wrote this. There was a reason, but I forget. I wanna say (99% sure) that most tests in the codebase had no clustername in the cluster update, as there was no reason to specify the reasoning as there was only one cluster callback persisted.

Reply:
99.99% sure now (haven't actually looked through the existing tests though, but rethinking about it I'm almost certain).